lib/services/storageManagement2/lib/operations/storageAccounts.js

```diff
@@ -39,10 +41,42 @@ function StorageAccounts(client) {
  * @param {object} [options.customHeaders] Headers that will be added to the
  * request
  *
- * @param {function} callback
+ * @returns {Promise} A promise is returned
+ * @resolve {HttpOperationResponse<CheckNameAvailabilityResult>} - The deserialized result object.
+ * @reject {Error} - The error object.
+ */
+StorageAccounts.prototype.checkNameAvailabilityWithHttpOperationResponse = function (name, options) {
+  let self = this;
+  let client = self.client;
+  return new Promise((resolve, reject) => {
+    self._checkNameAvailability(name, options, (err, result, request, response) => {
+      if (err) {
+        return reject(err);
+      }
+      let httpOperationResponse = new HttpOperationResponse(request, response);
+      httpOperationResponse.body = result;
+      return resolve(httpOperationResponse);
```

Comment:
Well I tested the code by removing the return keyword and added a `console.log()`. The result was amazing: `console.log()` got executed.

### Modified method by removing the "return" keyword 
```javascript
StorageAccounts.prototype.checkNameAvailability = function (name, options, optionalCallback) {
  var client = this.client;
  var self = this;
  if (!optionalCallback && typeof options === 'function') {
    optionalCallback = options;
    options = null;
  }
  if (!optionalCallback) {
    return new Promise((resolve, reject) => {
      self._checkNameAvailability(name, options, (err, result, request, response) => {
        if (err) reject(err);
        console.log('Before resolving...');
        console.log(result);
        resolve(result);  //<<<<<<
        console.log('After resolving...');
      });
    });
  } else {
    return self._checkNameAvailability(name, options, optionalCallback);
  }
};
```

### The test code that called the method
```javascript
msRestAzure.loginWithServicePrincipalSecret(spn, secret, domain, function (err, credentials) {
  var client = new storageManagementClient(credentials, subscriptionId);
  let name = 'foo'; 
  //calling the method in promise style
  client.storageAccounts.checkNameAvailability(name).then(function (result) {
    console.log('In the thenable waiting for the result of the resolved promise..');
    console.log(result);
  }).catch(function (err) {
    console.log(err);
  });
});
```

##### Output:
```bash
bash-3.2$ node test.js
Before resolving...
{ nameAvailable: false,
  reason: 'AlreadyExists',
  message: 'The storage account named foo is already taken.' }
After resolving... //<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
In the thenable waiting for the result of the resolved promise..
{ nameAvailable: false,
  reason: 'AlreadyExists',
  message: 'The storage account named foo is already taken.' }
```

##### Output after adding the "return" keyword back to the resolve statement:

```bash
bash-3.2$ node test.js
Before resolving...
{ nameAvailable: false,
  reason: 'AlreadyExists',
  message: 'The storage account named foo is already taken.' }
In the thenable waiting for the result of the resolved promise..
{ nameAvailable: false,
  reason: 'AlreadyExists',
  message: 'The storage account named foo is already taken.' }
bash-3.2$
```

### TLDR; return is required based on the above test.

What do you think?